teps/0115-tekton-catalog-git-based-versioning.md

```diff
@@ -0,0 +1,707 @@
+    - [Best Practices](#best-practices)
+  - [Alternatives](#alternatives)
+    - [One resource per Catalog](#one-resource-per-catalog)
+    - [Submodules in Catalogs](#submodules-in-catalogs)
+  - [References](#references)
+<!-- /toc -->
+
+# TEP-0115: Tekton Catalog Git-Based Versioning
+
+## Summary
+
+Tekton Catalog organization and versioning is directory-based which presents challenges in scaling and version bumping.
+In this TEP, we propose a git-based versioning and flat organization for Tekton Catalogs. This TEP builds on prior 
+work in [TEP-0003][tep-0003] and [TEP-0110][tep-0110], and is a blocker for [TEP-0079][tep-0079].
+
+## Motivation
+
+As of today, the Tekton Catalog organization and versioning is directory-based: versions are expressed as directories.
+
+The Catalog organization contract with directory-based versioning is:
+
+```bash
+# Optional: owner(s) and reviewer(s) of the resource (all versions)
+./{resource-type}/{resource-name}/OWNERS
+# The README of the resource (what is the resource about, list of versions, …)
+./{resource-type}/{resource-name}/README.md
+
+# The resource itself
+./{resource-type}/{resource-name}/{version}/{resource}.yaml
+
+# The README of the versioned resource (usage, …)
+./{resource-type}/{resource-name}/{version}/README.md
+
+# Optional: Test working samples.
+# Those samples would be used to run automated tests
+./{resource-type}/{resource-name}/{version}/tests/...
+# Optional: Addition samples.
+# Those samples would be used to run automated tests
+./{resource-type}/{resource-name}/{version}/samples/...
+```
+
+For example, the a Tekton Catalog with multiple `Tasks` and `Pipelines` is organized as such:
+
+```bash
+./task/
+  /argocd
+    /0.1
+      /README.md
+      /argocd.yaml
+      /samples
+      /tests
+    /0.2
+      /README.md
+      /argocd.yaml
+      /samples
+      /tests
+    /OWNERS
+  /golang-build
+    /0.1
+      /README.md
+      /golang-build.yaml
+      /samples
+      /tests
+    /0.2
+      /README.md
+      /golang-build.yaml
+      /samples
+      /tests
+    /OWNERS
+  /foo-task
+    /0.1
+    /0.2 
+  /bar-task
+    /0.1
+    /0.2
+    /0.3
+  /...
+./pipeline/
+  /go-release
+    /0.1
+      /README.md
+      /go-release.yaml
+      /samples
+      /tests
+    /0.2
+      /README.md
+      /go-release.yaml
+      /samples
+      /tests
+    /OWNERS
+  /foo-pipeline
+    /0.1
+    /0.2 
+  /bar-pipeline
+    /0.1
+    /0.2
+    /0.3
+  /...
+```
+
+For further information, see [TEP-0003: Tekton Catalog Organization][tep-0003].
+
+The current directory-based organization and versioning presents challenges:
+- Bumping versions of resources is cumbersome which encourages making "minor" changes in existing versions. Some changes
+that seem minor cause surprising breaking changes for users, such as described in [tektoncd/catalog#784][784].
+- Resources that are frequently updated with new versions end up with a lot of duplications because any change requires
```

Comment:
Grammar nit: "duplication" should be singular.

Reply:
(line 147)

Reply:
English 😅 

Fixed it, thanks!